### PR TITLE
modify files versions tests to pass with oC10 provider

### PR DESCRIPTION
**Pull request by individual-it**
part of #709

- fix matcher to be correct V3Matcher function call
- delete unneeded checks
- delete access control headers
- don't delete created file after tests
- make fileVersions tests to work with oc10 provider

### CODE REVIEW

**Comment by individual-it** [comment]
after this PR
on the consumer side:
image

on the provider side:
image

meaning one more test is run but the same amount of tests fail on the provider side

File: tests/fileVersionTest.js [hunk 1]
```diff
@@ -201,32 +179,62 @@ describe('Main: Currently testing file versions management,', function () {
...
-              done()
-            })
-          }).catch(reason => {
-            fail(reason)
-            done()
-          })
+        return oc.fileVersions.restoreFileVersion(fileInfo.id, fileInfo.versions[0].versionId, versionedFile).then(status => {
+          expect(status).toBe(true)
+        }).catch(reason => {
+          fail(reason)
```
Comment (dpakach) [inline tests/fileVersionTest.js:237]: why did you strip down this test?

Comment (individual-it) [reply tests/fileVersionTest.js:237]: this test is about restoring file version, no need to check the content here as we are testing against a mock server anyway see 86ae9a09984badee76c87525eb865c630135c900

File: tests/webdavHelper.js [hunk 2]
```diff
@@ -82,8 +89,37 @@ const deleteItem = function (user, password, itemName) {
...
+    method: 'PROPFIND',
+    body: '<?xml version="1.0"?>' +
+          '<d:propfind  xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns">' +
+          '<d:prop><oc:fileid /></d:prop>' +
+          '</d:propfind>',
+    headers: { authorization: getAuthHeaders(user, password) }
+  })
+  chai.assert.strictEqual(
+    fileIdResult.status, 207, `could not get fileId for '${itemName}'`
+  )
```
[the remark this answers is not in the document]
Comment (individual-it) [reply tests/webdavHelper.js]: fixed

File: tests/providerTest.js [hunk 3]
```diff
@@ -144,6 +133,22 @@ describe('provider testing', () => {
...
+      'file version link is returned': (setup, parameters) => {
+        if (setup) {
+          const fileId = getFileId(parameters.username, parameters.password, parameters.fileName)
+
+          const doc = new Dom().parseFromString(
+            listVersionsFolder(parameters.username, parameters.password, fileId)
+          )
+          const select = xpath.useNamespaces({ d: 'DAV:' })
+          const nodes = select('//d:response/d:href', doc)
+          const nodeValue = nodes[parameters.number].childNodes[0].nodeValue
```
[the remark this answers is not in the document]
Comment (individual-it) [reply tests/providerTest.js]: OK, even I like xpath sooo much I've changed it over

**Review by dpakach — APPROVED** [review]

**Pull request merged**